policy/modules/services/incus.te

```diff
@@ -0,0 +1,198 @@
+policy_module(incus)
+########################################
+#
+# Declarations
+#
+
+container_engine_domain_template(incusd)
+container_system_engine(incusd_t)
+type incusd_exec_t;
+container_engine_executable_file(incusd_exec_t)
+init_daemon_domain(incusd_t, incusd_exec_t)
+ifdef(`enable_mls',`
+	init_ranged_daemon_domain(incusd_t, incusd_exec_t, s0 - mls_systemhigh)
+')
+
+type incusc_t;
+type incusc_exec_t;
+
+container_engine_executable_file(incusc_exec_t)
+application_domain(incusc_t, incusc_exec_t)
+
+dev_rw_vhost(incusd_t)
+
+# other engines like docker have this, what will it do?
+#container_engine_domain_template(incusd_user)
+
+# user containers are not tested yet
+#container_user_engine(incusd_user_t)
+#application_domain(incusd_user_t, incusd_exec_t)
+
+# other engines like docker have this, what will it do?
+#mls_trusted_object(incusd_user_t)
+
+type incusc_user_t;
+application_domain(incusc_user_t, incusc_exec_t)
+
+########################################
+#
+# incus daemon local policy
+
+allow incusd_t self:process getattr;
+
+# incusd wants to load netdev-phys*
+kernel_request_load_module(incusd_t)
+
+ifdef(`init_systemd',`
+	init_start_system(incusd_t)
+	init_stop_system(incusd_t)
+	init_get_system_status(incusd_t)
+	init_stop_generic_units(incusd_t)
+')
+
+# comm="incusd" path="/run/systemd/resolve/stub-resolv.conf"
+# allow incusd_t container_tmpfs_t:file { open read };
+# allow incusd_t container_tmpfs_t:dir search;
+container_read_tmpfs_files(incusd_t)
+
+fs_getattr_binfmt_misc_fs(incusd_t)
+fs_getattr_binfmt_misc_dirs(incusd_t)
+fs_dontaudit_manage_fusefs_files(incusd_t)
+
+# here dontaudit would also work, but consuming more resources
+fs_watch_tmpfs_dirs(incusd_t)
+
+dnsmasq_domtrans(incusd_t)
+dnsmasq_signal(incusd_t)
+dnsmasq_kill(incusd_t)
+dnsmasq_signull(incusd_t)
+
+kernel_setsched(incusd_t)
+dpkg_dontaudit_manage_db(incusd_t)
+
+files_exec_usr_files(incusd_t)
+
+# no error, but this AVC:
+#  scontext="system_u:system_r:incusd_t:s0" #  tcontext="system_u:system_r:iptables_t:s0" #  class="process"
+#  perms="sigkill" #  comm="incusd" exe="" path=""
+# candidate for dontaudit instead?
+iptables_admin(incusd_t,system_r)
```

Comment:
If I try I still get:
```
#============= incusd_t ==============
# audit(1734635352.998:154972):
#  scontext="system_u:system_r:incusd_t:s0" tcontext="system_u:system_r:iptables_t:s0"         
#  class="process" perms="sigkill"
#  comm="incusd" exe="" path=""
#  message="type=AVC msg=audit(1734635352.998:154972): avc:  denied  { sigkill }
#   for  pid=1077723 comm="incusd" scontext=system_u:system_r:incusd_t:s0
#   tcontext=system_u:system_r:iptables_t:s0 tclass=process permissive=0 "
allow incusd_t iptables_t:process sigkill;

```

Reply:
We'll probably need to add new interfaces to `iptables` for these missing rules then. We can't use `iptables_admin()` because it requires passing in `system_r` which the `incus` module doesn't own.

It works, sure, but it's unpalatable for usptream.